### DISCUSSION TITLE
Simulation time and ending status.

### DISCUSSION THREAD

**Discussion by hazelPhd** [whitefield-framework/whitefield#125] [Q&A]
Hi,

I couldn't find where I can specify the time of the simulation? Also, when the simulation ends, is there a way to be notified that it ended instead of checking manually?

Dose the time calculation in Whitefield run as real-time? Or is it similar to the way Cooja does?

Thank you.

**Comment by nyrahul** [comment]
The time-calculation in Whitefield is in real-time. Thus for Whitefield, you do not have to explicitly pass the simulation time duration. Whenever, you think the time is up you can externally stop the Whitefield.

For e.g., to stop after 1 hour you can `sleep 3600 && ./scripts/wfshell stop_whitefield`

Hope it helps.

**Reply by hazelPhd** [reply]
This is the answer that I'm looking for. Thank you.

**Reply by hazelPhd** [reply]
<img width="860" alt="Screenshot 2021-08-09 at 5 31 59 pm" src="https://user-images.githubusercontent.com/51526498/128741370-269bd21b-e471-4add-8d67-a9aee32644d0.png">
I've been trying to simulate 400 nodes, but when I try and print the topology I only get 15 nodes, so I've tried to lower the number of nodes, but I got this error.
Is there a limit to the number of nodes? Do I have to wait for a while until the simulation gets set up if I have many nodes?

**Reply by nyrahul** [reply]
Can you share the `wf.cfg` and check the `whitefield/logs/*.log` files for any error?
Also, you can run the regression tests `./regression/regress.sh ./regression/basic` to check if anything is failing.

**Reply by hazelPhd** [reply]
The above image is from airline.log from `whitefield/logs/*.log `. I've ran the command `./regression/regress.sh ./regression/basic` and this is the result: 
`executing ./regression/basic/TC_two_contiki_nodes.test...
waiting for routing table status...
waiting 7/20 sec...
cmd_node_osname="contiki"
cmd_def_route=[NONE]
cmd_def_route=fe80::302:304:506:0
cmd_rpl_stats.resets=2
cmd_route_table=fd00::302:304:506:1 128 fe80::302:304:506:1
cmd_ipv6_stats: rcvd=4, sent=2
cmd_icmp_stats: rcvd=5, sent=2
cmd_config_info: rt_maxsz=100, nbr_maxsz:100
[1] [TC_two_contiki_nodes.test] [PASS] [tot_fail:0]
tc_msg:reached in 7 sec
executing ./regression/basic/TC_nodeexec_invalid_cfg.test...
[2] [TC_nodeexec_invalid_cfg.test] [PASS] [tot_fail:0]
executing ./regression/basic/TC_no_stackline.test...
[3] [TC_no_stackline.test] [PASS] [tot_fail:0]
executing ./regression/basic/TC_invalid_cfg.test...
[4] [TC_invalid_cfg.test] [PASS] [tot_fail:0]
executing ./regression/basic/TC_two_mixed_nodes.test...
[5] [TC_two_mixed_nodes.test] [FAIL] [tot_fail:1]
tc_msg:expecting return 0 instead got 0
check /whitefield/regression/testcase_report.log for execution log...
1/5 testcases failed!!
`

**Reply by nyrahul** [reply]
This looks like your basic system is setup ok (1 failure case seems to be a flaky regression). The problem seems to be with scale i.e., 400 nodes. Can you send me your updated `wf.cfg` and I can try to check at my end? What is the stackline you experimenting with ? RIOT/Contiki/Contiki-ng?

**Reply by hazelPhd** [reply]
'#key[start-end]=val ... Description, isMandatory?, supportsRange?, exampleValue

numOfNodes=169

#include=config/prop_model.cfg

#---------[Airline configuration]-------
#randSeed=0xabcdef #If not set every time a random seed is used
fieldX=400  #field space in x direction ... currently 2D model is supp only.
fieldY=400  #field space in y direction
topologyType=grid	#grid, randrect (ns3 RandomRectanglePositionAllocator),
gridWidth=5  #Grid topology width if the topologyType=grid
#nodePosition[1]=10,20,0 #Change the node position to the given vector
#nodePromiscuous[2]=1 #Sets promiscuous mode for node
#txPower=0 # Transmit Power (dBm)
#txPower[0]=7 # Transmit Power (dBm)

panID=0xabcd
NS3_captureFile=pcap/pkt
macPktQlen=20           #Maximum number of packets that can be outstanding on mac layer
macMaxRetry=3           #Max number of times the mac packet will be retried

'#---------[Stackline configuration]-------
# Format:
# nodeExec=path/to/bin $NODEID env1=abc env2=xyz
# nodeExec[node-range]=path/to/bin $NODEID env1=abc env2=xyz
#   where node-range could be 0, 0-10 etc'

nodeExec=thirdparty/contiki-ng/examples/rpl-udp/udp-client.whitefield $NODEID UDPCLI_SEND_INT=30 AUTO_START=1 UDP_PAYLOAD_LE$
#nodeExec="thirdparty/RIOT/tests/whitefield/bin/native/riot-whitefield.elf" -w $NODEID
nodeExec[0]=thirdparty/contiki-ng/examples/rpl-udp/udp-server.whitefield.elf $NODEID
#nodeExec[1]="thirdparty/RIOT/tests/whitefield/bin/native/nodes-ngwhitefield.elf" -w $NODEID'

**Reply by hazelPhd** [reply]
I'm using Contiki-ng. When I start Whitefield it says that it started OK, but when I execute the monitor script, it shows that Whitefield is stopped.
